core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchImpl.java

```diff
@@ -133,4 +142,51 @@ public String getLoadMoreButtonText() {
         .map(t -> t.get(SearchResultsImpl.I18N_KEY_LOAD_MORE_BUTTON_LABEL))
         .orElse(StringUtils.EMPTY);
   }
+
+  @Override
+  public int getAutocompleteTriggerThreshold() {
+    return AUTOCOMPLETE_THRESHOLD;
+  }
+
+  @Override
+  public List<String> getSearchTabs() {
+    return Optional.ofNullable(request)
+        .map(SlingHttpServletRequest::getResource)
+        .map(r -> r.getChild(NN_SEARCH_TABS_CONTAINER))
+        .map(r -> r.adaptTo(ResourceWrapper.class))
+        .map(ResourceWrapper::getDirectChildren)
+        .orElse(Stream.empty())
+        .map(this::getSearchTabUrl)
+        .filter(StringUtils::isNotBlank)
+        .collect(Collectors.toList());
+  }
+
+  private Set<Filter> getMergedFilters() {
+    Set<Filter> distinctFilters = new HashSet<>(getCaFilters());
+    if (filters != null) {
+      distinctFilters.addAll(filters);
+    }
+    return distinctFilters;
+  }
+
+  private List<Filter> getCaFilters() {
+    return Optional.ofNullable(request.getResource().adaptTo(ConfigurationBuilder.class))
+        .map(configurationBuilder -> configurationBuilder.as(SearchConfiguration.class))
+        .map(SearchConfiguration::searchFilters)
+        .map(Arrays::stream)
+        .orElse(Stream.empty())
+        .map(searchFilterConfiguration -> new FilterImpl(searchFilterConfiguration.name(),
+            searchFilterConfiguration.value()))
+        .collect(Collectors.toList());
+  }
+
+  private String getSearchTabUrl(@NonNull Resource searchTab) {
+    try {
+      return new URIBuilder(String.format("%s.%s.%s", searchTab.getPath(), ExporterConstants.SLING_MODEL_SELECTOR,
```

Comment:
this doesn't work with sling mappings, so might only work on author. 
we need to clarify how we want to do resource mapping

Reply:
- [ ] lets tackle this today